src/stdlib_math.fypp

```diff
@@ -6,14 +6,155 @@ module stdlib_math
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace, lnspace, EULERS_NUMBER_DP, EULERS_NUMBER_SP
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    real(dp), parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace 
+    real(sp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
```

Comment:
Should those constatns go in a specfic module, e.g.., `stdlib_constants`?
See, e..g., the issue #99

Reply:
I read through the issue and I don't feel that I am qualified to make the decision of starting `stdlib_constants` or establishing a `constants` naming precedence. In this particular instance, I just wanted `e` available when computing `lnspace`. These constants can easily get refactored and I think that step is more appropriate at a later time, either when a `stdlib_constants` module gets created or if the number of constants in use in `stdlib_math` starts to explode

Reply:
Yes, I suggest keep this here in this PR, and it can be moved to `stdlib_constants` in a separate PR.